src/main/webapp/WEB-INF/jsp/shoppingcart.jsp

```diff
@@ -551,71 +555,81 @@
+            var htmlTagId = 'qr';
 
-                        let httpUtil = new HttpUtil();
-                        httpUtil.sendPostRedirect("./transaction-commit.html", params);
-                    },
-                    canceled: function () {
-                        // callback rejected by user
-                        console.log("transacción cancelada por el usuario");
-                        onepay.drawQrImage("qr");
-                    },
-                    authorizationError: function () {
-                        // cacllback authorization error
-                        console.log("error de autorizacion");
-                    },
-                    unknown: function () {
-                        // callback to any unknown status recived
-                        console.log("estado desconocido");
-                    }
-                };
+            transaction["paymentStatusHandler"] = {
+                ottAssigned: function () {
+                    // callback transacción asinada
+                    console.log("Transacción asignada.");
+                    showLoadingImage();
+                },
+                authorized: function (occ, externalUniqueNumber) {
+                    // callback transacción autorizada
+                    console.log("occ : " + occ);
+                    console.log("externalUniqueNumber : " + externalUniqueNumber);
 
-                let onepay = new Onepay(transaction);
-                onepay.drawQrImage("qr");
-            },
-            error: function (data) {
-                console.log("something is going wrong");
-            }
+                    let params = {
+                        occ: occ,
+                        externalUniqueNumber: externalUniqueNumber
+                    };
+
+                    let httpUtil = new HttpUtil();
+                    httpUtil.sendGetRedirect("./transaction-commit.html", params);
+                },
+                canceled: function () {
+                    // callback rejected by user
+                    console.log("transacción cancelada por el usuario");
+                    Onepay.directQr(transaction, htmlTagId);
+                },
+                authorizationError: function () {
+                    // cacllback authorization error
+                    console.log("error de autorizacion");
+                },
+                unknown: function () {
+                    // callback to any unknown status recived
+                    console.log("estado desconocido");
+                }
+            };
+
+            Onepay.directQr(transaction, htmlTagId);
         });
     }
+
+    function doCheckout() {
+        var options = {
+            endpoint: './transaction-create.html',
+            commerceLogo: '/onepay-sdk-example/images/icons/logo-01.png',
+            callbackUrl: './transaction-commit.html'
+        };
+
+        Onepay.checkout(options, [{name:'param1',value:'value1'},{name:'param2',value:'value2'}]);
```

Comment:
What are those extra name:paramX things?

Reply:
@goncafa to be more clear: what’s the second argument meant to be?

Reply:
It is only a posible feature if you want to add some custom params to be set on the endpoint of transaction create. It is not tested yet and if you prefer i could delete from this line.

Reply:
Yeah, let’s remove them until it’s a documented feature